src/main/java/io/vertx/proton/impl/ProtonClientImpl.java

```diff
@@ -60,14 +61,43 @@ public void connect(ProtonClientOptions options, String host, int port,
   public void connect(ProtonClientOptions options, String host, int port, String username, String password,
                       Handler<AsyncResult<ProtonConnection>> handler) {
     final NetClient netClient = vertx.createNetClient(options);
-    connectNetClient(netClient, host, port, username, password, new ConnectCompletionHandler(handler), options);
+
+    NetClientConnectHandler connectHandler = new NetClientConnectHandler(netClient, host, username, password,
```

Comment:
Thats fair, reading it. Since it is already doing implicit SNI on the host unless a non-null value is passed for serverName, I think we should say null is valid and has the same effect, or that it isn't valid and throw accordingly.